### DISCUSSION TITLE
CSMF2.5 No Output

### DISCUSSION THREAD

**Discussion by sunnydays99** [NASA-LIS/LISF#1459] [Q&A] [answered]
Hello everyone,

I either get error messages for my CLSMF2.5 OL run that none of the variables in my MODEL_OUTPUT.TBL are available even through according to the source code it should be possible to output them. 
When I changed all names in the TBL file to lower case, the model runs and and I get no error messages, however all the generated output files are empty. 
If anyone has some advice or knows what is going wrong, I would gladly appreciate any help. The forcing variables, model output and lis.config and the ldt file are attached below. 

Kind regards and thank you so much.

[forcing_variables.txt](https://github.com/NASA-LIS/LISF/files/13428438/forcing_variables.txt)
[CLSM_OUTPUT_LIST.TBL.txt](https://github.com/NASA-LIS/LISF/files/13428452/CLSM_OUTPUT_LIST.TBL.txt)
[lis.config_clsm.ol.txt](https://github.com/NASA-LIS/LISF/files/13428526/lis.config_clsm.ol.txt)
[lis_input.gdas.clsm.nc.txt](https://github.com/NASA-LIS/LISF/files/13428535/lis_input.gdas.clsm.nc.txt)

**Comment by sunnydays99** [comment]
This is the error message for each undefined variable

Swnet                                              field is not defined
 [ERR] for diagnostic output...
 [ERR] Please exclude it from the model output attributes table
 [ERR] Program stopping ..
 [ERR] endrun is being called

**Comment by dmocko** [comment]
Hi @sunnydays99 

Can you please attach the lislog.0000 file that shows that variable(s) are unavailable?

If you use a different CLSM_OUTPUT_LIST.TBL file with the run that produces these message, please also attach the updated TBL file.

**Reply by sunnydays99** [reply]
Yes absolutely, here are my log files and they are run with the TBL file that I have already posted
[lislog_ol.0003.txt](https://github.com/NASA-LIS/LISF/files/13428897/lislog_ol.0003.txt)
[lislog_ol.0002.txt](https://github.com/NASA-LIS/LISF/files/13428898/lislog_ol.0002.txt)
[lislog_ol.0001.txt](https://github.com/NASA-LIS/LISF/files/13428899/lislog_ol.0001.txt)
[lislog_ol.0000.txt](https://github.com/NASA-LIS/LISF/files/13428900/lislog_ol.0000.txt)

**Reply by dmocko** [reply]
Thanks.  Not sure yet what's happening.  Can you also share your configure.lis file?  It's in LISF/lis/make.  And what system are you running this on?  The code should be case-insensitive, but from your description, it seems that the case makes a difference here.

**Reply by sunnydays99** [reply]
Here is the configure.lis file:
[configure.lis.txt](https://github.com/NASA-LIS/LISF/files/13429120/configure.lis.txt)

and the Linux distribution I am using is Ubuntu 22.04.3 LTS

**Reply by dmocko** [reply]
OK - I am able to reproduce your behavior.  And I think I know why it's happening.

Your lis_input file has all values of 0 for both LANDMASK and DOMAINMASK.  So, it's not able to write any output variables.

Can you please also attach your ldt.config file and your ldtlog file?

**Reply by sunnydays99** [reply]
okay I see that's what it causing the issue, thank you!

The ldt.config and log files are attached below:
[ldtlog.0000.txt](https://github.com/NASA-LIS/LISF/files/13430024/ldtlog.0000.txt)
[ldt.config.clsm_params.txt](https://github.com/NASA-LIS/LISF/files/13430025/ldt.config.clsm_params.txt)

**Reply by dmocko** [reply]
Thanks, I will try to work with this to see if I can get you a good lis_input file.

What endian option did you choose for the LDT configure step?

These .1gd4r files that you are trying to use are big_endian.

**Reply by sunnydays99** [reply]
That would be amazing! I compiled LDT with the big endian option

**Reply by dmocko** [reply, marked as answer]
I haven't had time to try to run with your ldt.config, but I noticed that these config entries need to be changed to:

```
Landmask lower left lat:       -59.875
Landmask lower left lon:      -179.875
Landmask upper right lat:       89.875
Landmask upper right lon:      179.875
```

The `landmask_mod44w_025.1gd4r` file has these dimensions, and you should not use those other values here.

I think this will solve your problem, as you are probably only reading the lower-left corner of this dataset, which would be over the ocean and not contain any land points.

From where did you get the CLSMF25 parameter files?  They need to be on the same exact landmask as your run domain.

**Reply by sunnydays99** [reply]
Hi thank you so much for your help!! I managed to fix the issue with the landmask file and when viewing the lis_input the landmask isn’t just zeros anymore. The lis run is now also yielding the desired output variables
